### ISSUE TITLE
More methods to work with uncorresponding bits

### ISSUE THREAD

**Issue by rusty-snake** [bitflags/bitflags#327]
- `complement` with `from_bits_retain`
  ```rust
  FlagsT::from_bits_retain(!FlagsT::all().bits())
  ```
- `contains_uncorresponding_bits` or `has_uncorresponding_bits`

**Comment by KodrAus** [comment]
Hi @rusty-snake :wave: Are you able to share any more details on what you're trying to do here?

**Comment by rusty-snake** [comment]
Yes, my use case are flags of ffi functions where I can only allow known safe flags. Passing known unsafe or flags added by future versions might be unsafe. Instead of silent truncating the bits I want to check and return an error.

**Comment by KodrAus** [comment]
Ahh, so you're suggesting a function to detect whether or not there are any bits set that don't correspond to a flag? Would it be enough to write?

```rust
let bits = Flags::from_bits(possibly_invalid.bits()).ok_or_else(|| "contains unrecognised bits")?;
```

**Comment by rusty-snake** [comment]
Yes, that would be an alternative way to implement it.

**Comment by KodrAus** [comment]
Hi @rusty-snake. Just calling out that `2.3.3` slightly alters the behavior of the `!` operator that affects flags with bits that only partially correspond to a known flag. I'm not sure if that will affect you, but wanted to draw your attention to it.

In https://github.com/bitflags/bitflags/issues/363#issuecomment-1608305273 we're trying to come up with a fully consistent treatment of unknown bits, which might become a `3.0.0` release. That would let us include some new inherent methods for detecting and truncating them if you wish.

**Comment by KodrAus** [comment]
In #400 we added _strict_ methods to the parser and formatter to work with flags that only want bit patterns that fully contain some number of defined flags. We could consider adding `from_bits_strict` as a variant that's even tighter than `from_bits` and `from_bits_truncate`. The difference only matters for multi-bit flags, but is outlined [here](https://github.com/bitflags/bitflags/blob/main/spec.md#truncate).

I've also found myself wanting a `.truncate()` inherent method on flags types too.

**Comment by KodrAus** [comment]
Just coming back through some triage, we've since added more methods to work with and unset unknown bits so I'll go ahead and close this one. Any more specific method suggestions can be made through new issues.
